app/experimenter/experiments/tests/test_forms.py

```diff
@@ -272,126 +264,369 @@ def test_form_is_invalid_if_population_percent_above_100(self):
         self.assertIn("population_percent", form.errors)
 
 
+class TestExperimentVariantsFormSet(TestCase):
+
+    def setUp(self):
+        self.FormSet = inlineformset_factory(
+            formset=ExperimentVariantsFormSet,
+            model=ExperimentVariant,
+            parent_model=Experiment,
+            fields=["is_control", "ratio", "name", "slug", "description"],
+        )
+
+        self.data = {
+            "variants-TOTAL_FORMS": "3",
+            "variants-INITIAL_FORMS": "0",
+            "variants-MIN_NUM_FORMS": "0",
+            "variants-MAX_NUM_FORMS": "1000",
+            "variants-0-is_control": True,
+            "variants-0-ratio": "34",
+            "variants-0-name": "control name",
+            "variants-0-slug": "control-slug",
+            "variants-0-description": "control desc",
+            "variants-1-is_control": False,
+            "variants-1-ratio": "33",
+            "variants-1-name": "branch 1 name",
+            "variants-1-slug": "branch-1-slug",
+            "variants-1-description": "branch 1 desc",
+            "variants-2-is_control": False,
+            "variants-2-ratio": "33",
+            "variants-2-name": "branch 2 name",
+            "variants-2-slug": "branch-2-slug",
+            "variants-2-description": "branch 2 desc",
+        }
+
+    def test_formset_valid_if_sizes_sum_to_100(self):
+        formset = self.FormSet(data=self.data)
+        self.assertTrue(formset.is_valid())
+
+    def test_formset_invalid_if_sizes_sum_to_less_than_100(self):
+        self.data["variants-0-ratio"] = 30
+        formset = self.FormSet(data=self.data)
+        self.assertFalse(formset.is_valid())
+
+    def test_formset_invalid_if_sizes_sum_to_more_than_100(self):
+        self.data["variants-0-ratio"] = 40
+        formset = self.FormSet(data=self.data)
+        self.assertFalse(formset.is_valid())
+
+    def test_formset_invalid_if_duplicate_slugs_appear(self):
+        self.data["variants-0-slug"] = self.data["variants-1-slug"]
+        formset = self.FormSet(data=self.data)
+        self.assertFalse(formset.is_valid())
```

Comment:
I wonder if it would make sense to also check that `total_error_count() == 1` for these in case future validation errors pop up that might make this pass but isn't only testing this specific use case.

Reply:
Oh yep I usually check for membership of the field in form.errors, I'll add that